src/main/java/it/smartcommunitylab/aac/files/store/LocalFileStore.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.smartcommunitylab.aac.files.store;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.UrlResource;
+
+public class LocalFileStore implements FileStore {
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	@Value("${persistence.files.path}")
+	private String basePath;
+	
+	private Path root;
+
+	public void setPath(String realm) {
+		this.root = Paths.get(basePath + System.getProperty("file.separator") + realm);
```

Comment:
remove the System.getProperty(), it's not this class concern where the value comes from. 
This should be set via StorageConfig.